server/src/main/java/io/spine/server/delivery/PickUpAckMixin.java

```diff
@@ -0,0 +1,114 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.delivery;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import io.spine.annotation.GeneratedMixin;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A mixin of the {@link PickUpAck} that provides convenient methods for accessing the result.
+ */
+@GeneratedMixin
+public interface PickUpAckMixin {
+
+    /**
+     * Creates a new {@code PickUpOutcome} of successfully picked shard
+     * with the given {@code ShardSessionRecord}.
+     */
+    @SuppressWarnings("ClassReferencesSubclass") // This is a mixin of the type.
+    static PickUpAck pickedUp(ShardSessionRecord session) {
+        checkNotNull(session);
+        return PickUpAck
+                .newBuilder()
+                .setSession(session)
+                .vBuild();
+    }
+
+    /**
+     * Creates a new {@code PickUpOutcome} indicating that the shard is already picked by the given
+     * {@code worker}.
+     */
+    @SuppressWarnings("ClassReferencesSubclass") // This is a mixin of the type.
+    static PickUpAck alreadyPickedBy(WorkerId worker) {
+        checkNotNull(worker);
+        return PickUpAck
+                .newBuilder()
+                .setAlreadyPickedBy(worker)
+                .vBuild();
+    }
+
+    /**
+     * Calls the given {@code consumer} with the {@code ShardProcessingSession} if the shard is
+     * successfully picked.
+     */
+    @CanIgnoreReturnValue
+    default PickUpAckMixin ifPicked(Consumer<ShardSessionRecord> consumer) {
+        if (getSession() != null) {
+            consumer.accept(getSession());
+        }
+        return this;
+    }
+
+    /**
+     * Calls the given {@code consumer} if the shard could not be picked because it's already
+     * picked by another worker.
+     *
+     * <p>The worker who owns the session will be passed to the {@code consumer}.
+     */
+    @CanIgnoreReturnValue
+    default PickUpAckMixin ifAlreadyPicked(Consumer<WorkerId> consumer) {
+        if (getAlreadyPickedBy() != null) {
+            consumer.accept(getAlreadyPickedBy());
+        }
+        return this;
+    }
+
+    /**
+     * Returns {@code ShardProcessingSession} if this outcome indicates that shard is successfully
+     * picked, or empty {@code Optional} otherwise.
+     */
+    default Optional<ShardSessionRecord> session() {
+        return Optional.ofNullable(getSession());
+    }
+
+    /**
+     * Returns {@code WorkerId} if this outcome indicates that shard could not be picked
+     * as it's already picked by another worker, or empty {@code Optional} otherwise.
+     */
+    default Optional<WorkerId> alreadyPickedBy() {
+        return Optional.ofNullable(getAlreadyPickedBy());
+    }
+
+    ShardSessionRecord getSession();
```

Comment:
Please document. And also, I think there may be some warnings by IDEA. See which annotations we use to mark these methods, in similar mixins.

Same below.